Remove "Reduction for Rvalues" from reference (#2050)
oneapi-src/oneTBB: doc/main/tbb_userguide/rvalue_reduce.rst

.. _rvalue_reduce:

Parallel Reduction for Rvalues

> ==============================

Reviewer: ```suggestion
Use of r-Values for Reduction
=============================
```

The functional form of ``parallel_reduce`` supports rvalue references in both the reduction
function and the combine function. This allows efficient use of move semantics when

> accumulating results that are expensive to copy, such as containers.

Reviewer: ```suggestion
function and the combine function. This allows efficient use of move operations when
accumulated results are expensive to copy.
```

The example below merges a collection of ``std::set`` objects into a single set.
Because the accumulator is passed as an rvalue reference, the algorithm can use
the C++17 ``std::set::merge`` function to transfer nodes between sets without copying or

> moving the underlying data.

Reviewer: ```suggestion
Because the accumulator is passed as an rvalue reference, the algorithm can
transfer nodes between sets without copying or moving the underlying data.
```
